lib/rpc.ts

```diff
@@ -301,11 +327,27 @@ export class Rpc extends EventEmitter implements IForwarderDest {
     return this.postMessageForward(msg.mdest || "", msg.data);
   }
 
+  // Mark outgoing messages for queueing.
+  private _queueOutgoing() {
+    if (!this._inactiveSendQueue) {
```

Comment:
I'm wondering wether an interface to check the state of the rpc could be useful. In particular while trying to recover from communication failure. May be something like `public isQueuingOutgoing(): boolean;` and `public isQueuingIncoming(): boolean;` `public isReadyReceived(): boolean;` , What do you think ?